src/web-server/app.py

```diff
@@ -49,35 +74,58 @@ def root():
     return app.send_static_file("./index.html")
 
 
-@app.route("/<path>")
+@app.route("/<path>/")
 def load_file(path):
-    raw = get_raw(path, True)
     filename = os.path.basename(path)
     if filename.startswith("IGNORE"):
         return None
-    if raw is None:
+
+    raw = load_shortlink_file(path)
+
+    if raw is NOT_LOGGED_IN:
+        response = redirect(url_for("login"))
+        response.set_cookie(COOKIE_SHORTLINK_REDIRECT, value=path)
+        return response
+    elif raw is NOT_FOUND:
         return app.send_static_file(path)
+    elif raw is NOT_AUTHORIZED:
+        return "This file is only visible to staff."
+
     data = bytes(
         json.dumps({"fileName": raw["full_name"], "data": raw["data"]}), "utf-8"
     )
 
     response = make_response(redirect("/", code=302))
-    response.set_cookie("load", value=data)
+    response.set_cookie(COOKIE_FILE_LOAD, value=data)
     return response
 
 
 @app.route("/<path>/raw")
-def get_raw(path, internal=False):
+def get_raw(path):
+    return jsonify(load_shortlink_file(path))
```

Comment:
`internal` was too confusing, so I broke `get_raw` up into a big helper function that does all the work of accessing the database, and a wrapper that provides an API for debugging.